library/core/src/hint.rs

```diff
@@ -121,7 +121,30 @@ pub fn black_box<T>(dummy: T) -> T {
     #[cfg(not(miri))] // This is just a hint, so it is fine to skip in Miri.
     // SAFETY: the inline assembly is a no-op.
     unsafe {
-        llvm_asm!("" : : "r"(&dummy));
+        #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+        {
+            asm!(
+                "/* {0} */",
+                in(reg) &dummy,
+                options(att_syntax, nostack, preserves_flags),
+            );
+        }
+        #[cfg(target_arch = "nvptx64")]
+        {
+            asm!(
+                "/* {0} */",
+                in(reg32) &dummy,
```

Comment:
NVPTX is a 64-bit architecture so pointers need to be in `reg64`, not `reg32`.

Reply:
Is this [table correct](https://doc.rust-lang.org/nightly/unstable-book/library-features/asm.html#register-operands) ?
![image](https://user-images.githubusercontent.com/15225902/93082475-319e5a00-f6bb-11ea-8b30-4aeb903e3fc4.png)

Reply:
Yes the table is correct. However `r` is only valid for use with pointers on 32-bit NVPTX. Our NVPTX target is 64-bit so you need to use `l` for pointers.

Reply:
Should we document NVPTX64 then ?